docs/content/docs/dev/datastream/fault-tolerance/checkpointing.md

```diff
@@ -271,4 +271,8 @@ Any operator that is prepared to be rescaled should work well with tasks that pa
 Restoring from a checkpoint where only a subset of tasks finished is equivalent to restoring such a
 task with the number of new subtasks equal to the number of finished tasks.
 
+### Waiting for the final checkpoint before task exit
+
+To ensure all the records could be committed for operators using the two-phase commit, the tasks would wait for the final checkpoint completed successfully after all the operators finished. It needs to be noted that this behavior would prolong the execution time of tasks. If the checkpoint interval is long, the execution time would also be prolonged largely. For the worst case, if the checkpoint interval is set to `Long.MAX_VALUE`, the tasks would in fact be blocked forever since the final checkpoint would never happen.
```

Comment:
```
To ensure all the records will be committed for operators using the two-phase commit, the tasks have to wait for the final checkpoint to complete successfully after all operators has finished processing new records. It is worth noting that this behavior prolongs the execution time of tasks. If the either checkpoint interval is long or end to end checkpoint duration is long, the execution time will also be prolonged significantly. In the worst case, if the checkpoint interval is set to `Long.MAX_VALUE`, the tasks would in fact be blocked forever since the final checkpoint will never happen.
```

+ can you split it into multiple lines? Docs that have paragraphs in a single line are causing quite a huge mess when reviewing the diffs or resolving merge conflicts.

Reply:
I split the lines now~